### DISCUSSION TITLE
[StopWatch] StopwatchEvent::getDuration() always returns 0

### DISCUSSION THREAD

**Discussion by lobodol** [symfony/symfony#47141] [Q&A]
### Symfony version(s) affected

6.0.4

### Description

Hello there,

I'm following the [official documentation](https://symfony.com/doc/current/components/phpunit_bridge.html#clock-mocking) in order to implement a time-sensitive unit test.

The problem is `StopwatchEvent::getDuration()` always returns 0.

### How to reproduce

My test:
```php
    /**
     * @group time-sensitive
     */
    public function testWatchForEndFlag(): void
    {
        $stopwatch = new Stopwatch();

        $stopwatch->start('fooEvent');
        sleep(10);
        $duration = $stopwatch->stop('fooEvent')->getDuration();

        $this->assertEquals(10000, $duration);
    }
```

Content of `phpunit.xml.dist`:
```xml
<?xml version="1.0" encoding="UTF-8"?>

<!-- https://phpunit.readthedocs.io/en/latest/configuration.html -->
<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:noNamespaceSchemaLocation="vendor/phpunit/phpunit/phpunit.xsd"
         backupGlobals="false"
         colors="true"
         bootstrap="tests/bootstrap.php"
         convertDeprecationsToExceptions="false"
>
    <php>
        <ini name="display_errors" value="1" />
        <ini name="error_reporting" value="-1" />
        <server name="APP_ENV" value="test" force="true" />
        <server name="SHELL_VERBOSITY" value="-1" />
        <server name="SYMFONY_PHPUNIT_REMOVE" value="" />
        <server name="SYMFONY_PHPUNIT_VERSION" value="9.5" />
    </php>

    <testsuites>
        <testsuite name="Project Test Suite">
            <directory>tests</directory>
        </testsuite>
    </testsuites>

    <coverage processUncoveredFiles="true">
        <include>
            <directory suffix=".php">src</directory>
        </include>
    </coverage>

    <listeners>
        <listener class="Symfony\Bridge\PhpUnit\SymfonyTestsListener" />
    </listeners>

    <!-- Run `composer require symfony/panther` before enabling this extension -->
    <!--
    <extensions>
        <extension class="Symfony\Component\Panther\ServerExtension" />
    </extensions>
    -->
</phpunit>
```

**Comment by lyrixx** [comment]
Since you are using `@group time-sensitive` all clock call are mocked. For exemple `sleep(10)` does not really sleep. That's why you see 0 here.

**Reply by lobodol** [reply]
Oh, I thought the annotation `@group time-sensitive`  was required to make clock mocking work. Actually, I understood that using this annotation would make `sleep(10)` pass instantly, but the `StopwatchEvent::getDuration()` return exactly 10 (not 10.00001 nor 0).

The example given in the documentation uses this annotation and still expects `getDuration()` to return 10 000ms:

```php
use PHPUnit\Framework\TestCase;
use Symfony\Component\Stopwatch\Stopwatch;

/**
 * @group time-sensitive
 */
class MyTest extends TestCase
{
    public function testSomething()
    {
        $stopwatch = new Stopwatch();

        $stopwatch->start('event_name');
        sleep(10);
        $duration = $stopwatch->stop('event_name')->getDuration();

        $this->assertEquals(10000, $duration);
    }
}
```

If I run it as it, the expectation fails:

```
Failed asserting that 0 matches expected 10000.
Expected :10000
Actual   :0
```

**Reply by lyrixx** [reply]
so you think, there is a bug somewhere? Can you create a reproducer?

**Reply by lobodol** [reply]
To reproduce, just have to create a new Symfony project, add the `MyTest` test and run it. IMO, either there is a bug somewhere or the documentation needs to be updated

**Reply by lyrixx** [reply]
OK, Feel free to submit a PR if you find something that can be improved :) Thanks

**Reply by lobodol** [reply]
Sure, I will if I can find a blatant bug.  
For now, I am surprised that nobody else faced this. When you run the code above, does the test pass without actually waiting 10 sec?
